src/Disks/DiskRestartProxy.cpp

```diff
@@ -24,6 +24,8 @@ class RestartAwareReadBuffer : public ReadBufferFromFileDecorator
 
     void setReadUntilPosition(size_t position) override { impl->setReadUntilPosition(position); }
 
+    void setReadUntilEnd() override { impl->setReadUntilEnd(); }
```

Comment:
Main fix here?

Reply:
No, 50/50 with other code. There we two different exceptions.